frontend/src/components/maps/PopupView.tsx

```diff
@@ -32,18 +29,22 @@ export const PopupView: React.FC<IPopupViewProps> = ({
   const { getParcel, getBuilding } = useApi();
   const [property, setProperty] = React.useState<IProperty | null>(propertyDetail);
   const id = propertyDetail?.id;
+  const agencyId = propertyDetail?.agencyId;
+  const propertyTypeId = propertyDetail?.propertyTypeId;
 
   React.useEffect(() => {
-    if (propertyTypeId === PropertyTypes.PARCEL) {
-      getParcel(id as number).then(parcel => {
-        setProperty(parcel);
-      });
-    } else if (propertyTypeId === PropertyTypes.BUILDING) {
-      getBuilding(id as number).then(building => {
-        setProperty(building);
-      });
+    if ((agencyId as number) > 0) {
```

Comment:
Only make a request to fetch more information if the property is owned by the user's agency or sub-agencies.